### DISCUSSION TITLE
Mathematical relationship between DFN/SPM models to Equivalent circuit model

### DISCUSSION THREAD

**Discussion by Akila1993** [pybamm-team/PyBaMM#3080] [Q&A]
Hi, 

Are there notebooks available where I can understand the mathematical relationship between the DFN/SPM  to ECM models available in PyBaMM?  I wanted to obtain the ECM for a particular DFN/SPM model.

Thank you.

**Comment by brosaplanella** [comment]
Here are two notebooks that explain the models

- DFN: https://docs.pybamm.org/en/latest/source/examples/notebooks/models/DFN.html
- SPM: https://docs.pybamm.org/en/latest/source/examples/notebooks/models/SPM.html

Additionally, we wrote this paper that clarifies the mathematical aspects of the different models: https://iopscience.iop.org/article/10.1088/2516-1083/ac7d31

**Reply by Akila1993** [reply]
@brosaplanella Thank you for your response. I was asking about the mathematical relationship of the **equivalent circuit model** to the DFN or SPM models. 

In other words how to obtain **equivalent circuit model** when the DFN model is available ?

Thank you.

**Reply by TomTranter** [reply]
AFAIK there isn't a direct way to transform the equations. You would have to cycle the DFN or SPM model and extract equivalent parameters in the same way you would a real battery

**Comment by chtld** [comment]
It just two types models. The Equivalent circuit model we only focus on the information about the Voltage and Current, and need to identification the parameter in the model, it based on the data. DFN and SPM is the model based on physics.
